### DISCUSSION TITLE
Question About Routing Table

### DISCUSSION THREAD

**Discussion by kmc7468** [ANLAB-KAIST/KENSv3#248] [Project Q&A 3. Internet Layer (IP/Routing)] [answered]
![image](https://github.com/ANLAB-KAIST/KENSv3/assets/18364980/3f6617d7-ba18-4f09-82bc-d099addf1377)

Consider the above topology. `R1` knows that the existence of the red ports (i.e. `R1` knows that 192.168.0.1 and 192.168.0.2 are vaild IP address.) However, I think `R1` doesn't know the existence of the blue ports, but it just knows the existence of `Subnet 2`, `Subnet 3` and `Subnet 4` thanks to the LSU packets from `R2`, `R3` and `R4`.

In this situation, what is the query result for 192.168.0.4, 192.168.0.5, 192.168.0.8 and 192.168.0.16 in the perspective of `R1`? I think the result for 192.168.0.4 and 192.168.0.5 should be 3 (`R1` -> `R2`) and the result for 192.168.0.8 should be 3 (`R1` -> `R2`) and the result for 192.168.0.16 should be 10 (`R1` -> `R2` -> `R3`). Am I right? If I'm wrong, is there any way for `R1` to know the existence of the blue ports?

**Comment by Pusnow** [comment, marked as answer]
Yes there is some ambiguity here.
You implement this in your own way.

However, you can know exact place of "router_id". We will grade your implementation with router_ids only.

**Reply by kmc7468** [reply]
Then if `R4`'s router id is 192.168.0.8, then should the query result be 11? (`R1` -> `R2` -> `R3` -> `R4`)

Also, I wonder how routing table is generated in reality. If we use real OSPF, what router table does `R1` have?

**Reply by Pusnow** [reply]
Yes.

For the second question, honestly I don't know. Feel free to consult RFCs.
